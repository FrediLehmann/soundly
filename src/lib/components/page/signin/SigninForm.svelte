<script type="ts" context="module">
  import { Button, Input } from '$lib/components/atoms';
</script>

<script type="ts">
  export let signin: svelte.JSX.FormEventHandler<HTMLFormElement>;
  export let submitting: boolean;
</script>

<form on:submit|preventDefault={signin}>
  <Input
    required
    type="email"
    name="email"
    label="Email"
    title="Please enter your email."
    disabled={submitting} />
  <Input
    required
    type="password"
    name="password"
    label="Password"
    title="Please enter your password."
    disabled={submitting} />
  <Button disabled={submitting} type="submit" style="primary" class="mbs-2">
    Sign in
  </Button>
</form>

<style>
  form {
    display: flex;
    flex-direction: column;
    gap: 1rem;
  }
</style>
