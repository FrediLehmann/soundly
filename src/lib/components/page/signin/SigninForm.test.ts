/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import SigninForm from './SigninForm.svelte';

describe('<SigninForm /> component', () => {
  test('renders correctly', () => {
    const { component } = render(SigninForm, {
      props: { signin: (_: unknown) => _, submitting: false }
    });
    expect(component).toMatchSnapshot();
  });
});
