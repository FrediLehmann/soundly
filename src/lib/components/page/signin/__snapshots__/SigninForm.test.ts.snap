// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SigninForm /> component renders correctly 1`] = `
SigninForm {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      false,
      "",
      "",
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "signin": 5,
      "submitting": 0,
    },
    "root": <div>
      <form
        autocomplete="on"
        class="svelte-1i4vv2l"
        name="Signin"
      >
        <div
          class="container svelte-1ulq3bj"
          data-required="true"
          data-sr-only="false"
        >
          <label
            class="svelte-1ulq3bj"
            for="email"
          >
            Email
          </label>
           
          <input
            class="svelte-1ulq3bj"
            id="email"
            required=""
            title="Please enter your email."
            type="email"
          />
           
        </div>
         
        <div
          class="container svelte-1ulq3bj"
          data-required="true"
          data-sr-only="false"
        >
          <label
            class="svelte-1ulq3bj"
            for="password"
          >
            Password
          </label>
           
          <input
            class="svelte-1ulq3bj"
            id="password"
            required=""
            title="Please enter your password."
            type="password"
          />
           
        </div>
         
        <button
          class="mbs-2 svelte-1dsjpe6"
          data-type="primary"
          type="button"
        >
          Sign in
        </button>
        
         
        <input
          hidden=""
          id="Signin_submit"
          type="submit"
        />
      </form>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
